### DISCUSSION TITLE
Handling of state locking in case of terraform job ending abruptly

### DISCUSSION THREAD

**Discussion by sahil1105** [kuptan/terraform-operator#42] [Q&A]
Hi, this looks like a great tool, so thanks for building it.
I had a question regarding how state locking is handled in case of failures. For context, an issue I've often seen is that when a terraform job is using a backend such as S3+DynamoDB, if the terraform execution is abruptly cancelled (say a K8s node goes down, or a pod needs to be moved to a different node during scale in), and we try to re-run anything, it gives an error saying that the state is locked and needs to be manually unlocked before execution can be resumed.
I wanted to know how this operator handles such situations?

**Comment by IbraheemAlSaady** [comment]
@sahil1105 thanks for opening the discussion 
This use case is not covered in this operator. I'm afraid that if we support it, it will introduce some issues that will mess with the state file since it needs to be forced unlock. My initial thought on this is that such case should be handled either manually, or by a separate workflow

I'm happy to discuss further if you have other thoughts on this
